reader.go

```diff
@@ -85,6 +90,18 @@ func (this *Reader) Read(p []byte) (n int, err error) {
 
 	// if we experienced an iconv error, check it
 	if err != nil {
+		// EINVAL:
+		// An incomplete multibyte sequence is encountered in the input,
+		// and the input byte sequence terminates after it.
+		if err == syscall.EINVAL {
```

Comment:
Assume an input stream which return a single byte in fillBuffer(), in this case, Read() will return an error, but actually this is not an error, so we must try to pull more data and return 
error only if there is no more new data. 

Moreover, in case of EINVAL, there is remaining data in buffer and fillBuffer() will not be then we will loop forever.

IMHO, the simplest way to deal with EINVAL is just try to read more data to determine whether this is an actual error.